src/ArrayObject.php

```diff
@@ -451,7 +451,7 @@ public function __isset($name)
      *
      * @throws \Rexlabs\ArrayObject\Exceptions\InvalidOffsetException
      */
-    public function offsetGet($offset)
+    public function offsetGet($offset): mixed
```

Comment:
I see that we are dropping support for php 7. I think it's because of this change as that's only supported from php 8. I am wondering if that is strictly required here.

Are we looking at just going with php 8 migration for wings without running in parallel with 7.4 like we did with the previous upgrade? I am not against it, just trying to understand that context for this change 🙂

Reply:
PHP 8 fatals or warns can't recall due to not matching the underlying interface return types.

mixed was only introduced in php 8. 

If you remove this the tests pass in php 7 but fail in all versions of php 8.

If you add this, the tests pass in php 8 but fail in all versions of php 7.

You could maybe? do this as a two stage release:

* Release 1: PHP 7.4 without this change with tests only for 7.4 tag as 4.0.0
* Release 2: PHP 8 (drop PHP 7.4) with this change with tests only for 8 tag as 4.1.0

Reply:
I am happy for us to go all in with php 8. Thank you for the explanation 🙏

Reply:
Going to try for the two stage release.

Reply:
2 stage release not needed - https://github.com/rexlabsio/array-object-php/pull/14#discussion_r1268045515